gcloud-java-storage/src/test/java/com/google/gcloud/storage/ITStorageTest.java

```diff
@@ -0,0 +1,470 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.storage;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import com.google.common.collect.ImmutableList;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Calendar;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class ITStorageTest {
+
+  private static StorageOptions options;
+  private static Storage storage;
+  private static RemoteGcsHelper gcsHelper;
+  private static String bucket;
+
+  private static final String CONTENT_TYPE = "text/plain";
+  private static final byte[] BLOB_BYTE_CONTENT = {0xD, 0xE, 0xA, 0xD};
+  private static final String BLOB_STRING_CONTENT = "Hello Google Cloud Storage!";
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @BeforeClass
+  public static void beforeClass() {
+    gcsHelper = RemoteGcsHelper.create();
+    if (gcsHelper != null) {
+      options = gcsHelper.options();
+      storage = StorageFactory.instance().get(options);
+      bucket = gcsHelper.bucket();
+      storage.create(BucketInfo.of(bucket));
+    }
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    if (storage != null) {
+      for (BlobInfo info : storage.list(bucket)) {
+        storage.delete(bucket, info.name());
+      }
+      storage.delete(bucket);
+    }
+  }
+
+  @Before
+  public void beforeMethod() {
+    org.junit.Assume.assumeNotNull(storage);
+  }
+
+  @Test
+  public void testListBuckets() {
+    ListResult<BucketInfo> bucketList = storage.list(Storage.BucketListOption.prefix(bucket));
```

Comment:
because of eventual consistency listing test would need to be in a loop with delay (failing if longer than X and still didn't reach objectives).

Same for object listing.